clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -2847,6 +2847,45 @@ def ByteSwapOp : CIR_BitOpBase<"byte_swap", CIR_UIntOfWidths<[16, 32, 64]>> {
   }];
 }
 
+//===----------------------------------------------------------------------===//
+// RotateOp
+//===----------------------------------------------------------------------===//
+
+def RotateOp : CIR_Op<"rotate", [Pure, SameOperandsAndResultType]> {
```

Comment:
Why doesn't this derive from CIR_BitOpBase? I'm not clear what we gain by having anything derive from that, but it seems like this should if other do.

Reply:
I believe the goal was to unify the assembly format and traits for all pure unary operations of the form `T -> T` in a single place. However, the name `BitOpBase` might misleadingly suggest that it specifically involves some bit-level semantics.

Formally, such operations are endomorphisms. That said, it's unclear to me whether names like `EndomorphismOpBase` or `EndoOpBase` are desirable or intuitive enough for this purpose.